File: solutions/golang/maximumSubarray/solution_test.go
package main

import "testing"

func TestMaxSubArray(t *testing.T) {
  tests := []struct {
    nums []int
    ans  int
  }{
    {
      nums: []int{-2, 1, -3, 4, -1, 2, 1, -5, 4},
      ans:  6,
    },
    {
      nums: []int{-2},
      ans:  -2,
    },
  }
  for _, tt := range tests {
    actual := maxSubArray(tt.nums)
    if actual != tt.ans {
      t.Errorf("got %d for input %v; expected %d",
        actual, tt.nums, tt.ans)
    }
  }
}

File: solutions/cpp/integerBreak/solution_test.cpp
//
// Created by mingyi on 26.05.20.
//


#include "gtest/gtest.h"

#include "solution.cpp"
#include "../include/structures.hpp"

using namespace std;
using namespace testing;

typedef TestStruct<int, int> TStruct;

class IntegerBreakTest : public ::testing::TestWithParam<TStruct> {};

TEST_P(IntegerBreakTest, Test
) {
  TStruct tt = GetParam();
  int actual = integerBreak(tt.input);
  ASSERT_EQ(tt
                .expected, actual);
}

INSTANTIATE_TEST_SUITE_P(Tests, IntegerBreakTest, ::testing::Values(
    TStruct{
        .input = 2,
        .expected =1
    },
    TStruct{
        .input = 10,
        .expected = 36
    }
)
);